drivers/onewire/onewire.c

```diff
@@ -0,0 +1,181 @@
+/*
+ * Copyright (C) 2016-2020 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser General
+ * Public License v2.1. See the file LICENSE in the top level directory for more
+ * details.
+ */
+
+/**
+ * @ingroup     drivers_onewire
+ * @{
+ *
+ * @file
+ * @brief       1-Wire bus driver implementation
+ *
+ * @author      Hauke Petersen <hauke.petersen@fu-berlin.de>
+ *
+ * @}
+ */
+
+#include <string.h>
+
+#include "assert.h"
+#include "xtimer.h"
+#include "onewire.h"
+
+#define ENABLE_DEBUG        (0)
+#include "debug.h"
+
+#define T_RESET_HOLD        (480)
+#define T_RESET_SAMPLE      (70)
+
+#define T_RW_PULSE          (3)
+#define T_W_HOLD            (57)
+#define T_W_RECOVER         (10)
+#define T_R_SAMPLE          (7)
+#define T_R_RECOVER         (55)
+
+static inline int _read_bit(const onewire_t *owi)
+{
+    gpio_clear(owi->pin);
+    xtimer_usleep(T_RW_PULSE);
+    gpio_init(owi->pin, owi->imode);
+    xtimer_usleep(T_R_SAMPLE);
+    int in = (gpio_read(owi->pin)) ? 1 : 0;
```

Comment:
```suggestion
    int in = gpio_read(owi->pin) ? 1 : 0;
```

Reply:
I like it better with those additional parenthesis in place as it is IMHO easier to apprehend the conditional meaning of that function call...